src/snapshot/snapshot.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package snapshot;
+
+import "google/protobuf/timestamp.proto";
+import "protoextensions/validate.proto";
+
+option go_package = "github.com/pachyderm/pachyderm/v2/src/snapshot";
+
+service API {
+  rpc CreateSnapshot(CreateSnapshotRequest) returns (CreateSnapshotResponse) {}
+  rpc DeleteSnapshot(DeleteSnapshotRequest) returns (DeleteSnapshotResponse) {}
+  rpc InspectSnapshot(InspectSnapshotRequest) returns (InspectSnapshotResponse) {}
+  rpc ListSnapshot(ListSnapshotRequest) returns (stream ListSnapshotResponse) {}
+}
+
+message CreateSnapshotRequest {
+  map <string, string> metadata = 1 [(validate.rules).map.keys.string.min_len = 1];
+}
+message CreateSnapshotResponse {
+  int64 id = 1 [(validate.rules).int64.gte = 1];
+}
+
+message DeleteSnapshotRequest {
+  int64 id = 1  [(validate.rules).int64.gte = 1];
+}
+message DeleteSnapshotResponse {}
+
+message SnapshotInfo {
+  int64 id = 1;
+  map <string, string> metadata = 2;
+  int64 chunkset_id = 4;
```

Comment:
There's no need to skip 3.  In the design doc it's hard to keep fields numbers up to date, but when copying it into the code, you can renumber them.